### DISCUSSION TITLE
Is there a way to let anyone manually trigger a workflow?

### DISCUSSION THREAD

**Discussion by dsaffo** [community/community#63336] [Actions] [answered]
### Select Topic Area

Question

### Body

I created a GitHub action that generates a PR from a JSON object that contributors generate from my website. The problem is it looks like you need to be a collaborator on the repo already before you can manually trigger workflows. Is there any way to allow anyone to trigger workflows?

**Comment by nirav-gajera** [comment, marked as answer]
Yes, there is a way to allow anyone to trigger workflows. You can do this by enabling the **Allow public workflows** setting for your repository. This setting allows anyone to trigger workflows in your repository, even if they are not collaborators.

To enable the **Allow public workflows** setting, follow these steps:

1. Go to the repository that you want to enable the setting for.
2. Click on the **Settings** tab.
3. Scroll down to the **Workflows** section and click on the **Allow public workflows** checkbox.
4. Click on the **Save changes** button.

Once you have enabled the **Allow public workflows** setting, anyone will be able to trigger workflows in your repository by sending a POST request to the following URL:

```
https://api.github.com/repos/OWNER/REPO/actions/workflows/WORKFLOW_ID/dispatches
```

Where:

* `OWNER` is the name of the organization or user that owns the repository.
* `REPO` is the name of the repository.
* `WORKFLOW_ID` is the ID of the workflow that you want to trigger.

For example, to trigger the workflow with the ID `123456789`, you would send the following POST request:

```
curl -X POST https://api.github.com/repos/OWNER/REPO/actions/workflows/123456789/dispatches
```

This will trigger the workflow and it will run as if it was triggered by a collaborator.

Please note that this setting can be dangerous if you are not careful. If you enable it, anyone will be able to trigger workflows in your repository, even if they are malicious. You should only enable this setting if you are sure that you trust everyone who will be able to trigger workflows.

**Reply by dsaffo** [reply]
Thanks for this information! Is there a way to pass an input to the workflow with this post request? In my case it would be a stringified version of the json.

**Reply by nirav-gajera** [reply]
yeah , you can pass an input to the workflow with the POST request. You can do this by setting the `inputs` parameter in the request body. The `inputs` parameter is a JSON object that contains the inputs that you want to pass to the workflow. For example, to pass the stringified version of the JSON object to the workflow, you would set the `inputs` parameter to the following:

```
{
  "json": "STRINGIFIED_JSON_OBJECT"
}
```

The workflow will then be able to access input in the `json` variable.

the example of a POST request you can use to trigger workflow and pass input:

```
curl -X POST https://api.github.com/repos/OWNER/REPO/actions/workflows/123456789/dispatches \
-H "Accept: application/vnd.github.antiope-preview+json" \
-H "Authorization: Bearer TOKEN" \
-d '{
  "inputs": {
    "json": "STRINGIFIED_JSON_OBJECT"
  }
}'
```

Where:

* `TOKEN` is a personal access token with the `repo` scope.
* `STRINGIFIED_JSON_OBJECT` is the stringified version of JSON object.

**Reply by dsaffo** [reply]
Thank you!

**Reply by EresDev** [reply]
I don't have the option "Allow public workflows". Why?

**Reply by reeew35** [reply]
Has github removed this option? I can't find it just like EresDev
